homeassistant/components/template/trigger_entity.py

```diff
@@ -48,6 +48,7 @@ def _process_data(self) -> None:
             **(run_variables or {}),
         }
 
+        self._render_availability_template(variables)
         self._render_templates(variables)
```

Comment:
Why do we need to call both methods? The `_render_templates` method anyway checks the state of availability template rendering, I think that method can also render the availability template?

Reply:
I think that makes sense to have `render_templates` to directly call the `render_availability_template`